### DISCUSSION TITLE
Shared procedure in multiple workspace

### DISCUSSION THREAD

**Discussion by arifpavel** [mit-cml/blockly-plugins#39] [Q&A]
Hi,
I wanted to implement a shared procedure to support creating procedure definitions in the workspace & having call procedures in a different workspace.
I have an idea to create/modify the call procedure block to get the procedures from separate storage rather than from the workspace procedure database.
Any other ideas or if anyone already implemented something similar to my use case?
Any help will be greatly appreciated.
Thanks

**Comment by mark-friedman** [comment]
@arifpavel In case you weren't aware, there is an existing Blockly plugin for shared procedures [here](https://www.npmjs.com/package/@blockly/block-shareable-procedures) and, in fact, it is the recommended way to include procedure blocks in Blockly (see doc [here](https://developers.google.com/blockly/guides/create-custom-blocks/procedures/overview)).

**Reply by arifpavel** [reply]
Hi @mark-friedman 
Thanks for your response. Yes, I know that. Just wanted to have lexical variable functionalities on top of that & this plugin already does that. Thought to make a workaround instead of reinventing the wheel. I will give it a try or use the [recommended plugin ](https://www.npmjs.com/package/@blockly/block-shareable-procedures) 
Thanks

**Reply by mark-friedman** [reply]
Ah, yes.  It would be nice to have the lexical variable version of the procedure blocks support the shareable procedure block plugin.  There is some information how to go about that in the Blockly documentation, [here](https://developers.google.com/blockly/guides/create-custom-blocks/procedures/creating-custom-procedure-blocks).  

@arifpavel Perhaps you'd like to give that a try as a pull request?

**Reply by arifpavel** [reply]
@mark-friedman Sure, I will try.
